Add performance tips

josh4trunks opened the pull request:
Feel free to make suggestions or edits. This is a work in progress.

RealRancor commented:
Hi,

another suggestion is to add a short note for windows to use 127.0.0.1 instead of localhost in the config.php db setting to speed up the database connection:

http://forum.owncloud.org/viewtopic.php?f=17&t=7559

jospoortvliet commented:
@RealRancor as I don't really know how that works - any chance you can propose some text?

josh4trunks commented:
That seems like pretty specific advice on a particular piece of software. If we're going to add that level of detail I also suggest we add advice on Unix systems to use a socket connection instead of a TCP connection for the database.

RealRancor commented:
@josh4trunks 
When "localhost" is used MySQL is doing socket connections instead of TCP (See https://dev.mysql.com/doc/refman/5.6/en/connecting.html)

@jospoortvliet 
Not quite sure how to describe this correctly but its just a replacement of:

```
'dbhost' => 'localhost',
```

to

```
'dbhost' => '127.0.0.1',
```

in the config.php

josh4trunks commented:
@RealRancor
localhost hat has nothing to do with the Unix **file** sockets I was proposing. I just meant using Unix file sockets would be another tip, similar in complexity and benefit as the one you were proposing.

Let's just get the sqlite note written up, and commit this thing. This way it'll be easier for people to submit PRs. @RealRancor if you want to write up your note we can try adding it now.

josh4trunks commented:
For the sqlite explanation, how about we link to here http://www.sqlite.org/whentouse.html
There's no anchors, but we could just reference/quote the "Situations Where Another RDBMS May Work Better". I think the main reason ownCloud works better on mysql is because of the "High Concurrency" point.

LukasReschke commented:
> I think the main reason ownCloud works better on mysql is because of the "High Concurrency" point.

That assumption is correct.

josh4trunks commented:
k, I'm on my smart phone, so either someone take this over and commit it, or I can add that note tonight.

RealRancor commented:
@josh4trunks
Not directly related to this PR but another quote for the localhost vs socket:

```
Note

Whenever you specify "localhost" or "localhost:port" as server, the MySQL client library will override this and try to connect to a local socket (named pipe on Windows). If you want to use TCP/IP, use "127.0.0.1" instead of "localhost".
```

from https://dev.mysql.com/doc/apis-php/en/apis-php-function.mysql-connect.html

josh4trunks commented:
Replaced by https://github.com/owncloud/documentation/pull/755
Sorry, I deleted the fork this was based off so I couldn't find a way to edit this PR anymore.
